celements-keycloak/src/main/java/com/celements/keycloak/KeycloakService.java

```diff
@@ -41,18 +45,42 @@ public class KeycloakService implements IdentityService {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(KeycloakService.class);
 
+  private static final String CELEMENTS_KEYCLOAK_REALM = "celements.keycloak.realm";
+
   private final Map<String, AuthenticationManager> authManagerCache = new ConcurrentHashMap<>();
+  private final Map<String, JwtDecoder> jwtDecoderCache = new ConcurrentHashMap<>();
 
   private final ConfigurationSource configSource;
+  private final Execution execution;
 
   @Inject
   public KeycloakService(
-      @Named(CelementsFromWikiConfigurationSource.NAME) ConfigurationSource configSource) {
+      @Named(CelementsFromWikiConfigurationSource.NAME) ConfigurationSource configSource,
+      Execution execution) {
     this.configSource = configSource;
+    this.execution = execution;
     LOGGER.info("KeycloakService constructor: {} host={}, realm={}", configSource.getClass(),
         getHost(), getRealm());
   }
 
+  @Override
+  public boolean isConfigValid() {
+    return configSource.containsKey(CELEMENTS_KEYCLOAK_REALM)
+        && getRealmOpt().map(realm -> !realm.trim().isEmpty()).orElse(false);
+  }
+
+  @Override
+  @NotEmpty
+  public Optional<String> getRealmOpt() {
```

Comment:
```suggestion
  @NotNull
  public Optional<@NotEmpty String> getRealmOpt() {
```
the optional may in fact be empty